tools/walletextension/multi_acc_helper.go

```diff
@@ -224,3 +224,22 @@ func executeCall(client *rpc.EncRPCClient, req *rpcRequest, resp *interface{}) e
 
 	return client.Call(resp, req.method, req.params...)
 }
+
+// The enclave requires the `from` field to be set so that it can encrypt the response, but sources like MetaMask often
+// don't set it. So we check whether it's present; if absent, we walk through the arguments in the request's `data`
+// field, and if any of the arguments match our viewing key address, we set the `from` field to that address.
+func setCallFromFieldIfMissing(args []interface{}, account common.Address) ([]interface{}, error) {
```

Comment:
This was previously in the main `wallet_extension.go` file, but was only used here.